modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -911,15 +1006,25 @@ private byte[] _getLogoBytes(
 			boolean useAccountEntryDefault)
 		throws Exception {
 
-		Long logoId = account.getLogoId();
+		Long logoId = GetterUtil.getLong(account.getLogoId());
 
-		if ((accountEntry != null) && (logoId == null) &&
-			useAccountEntryDefault) {
+		if (logoId == 0) {
+			FileEntry fileEntry =
+				_dlAppLocalService.fetchFileEntryByExternalReferenceCode(
+					contextCompany.getGroupId(),
+					account.getLogoExternalReferenceCode());
 
-			logoId = accountEntry.getLogoId();
+			if (fileEntry != null) {
+				return _file.getBytes(fileEntry.getContentStream());
```

Comment:
why here we cannot just assign the logoId = fileEntry.getFileEntryId() ? 
So then we do the same logic in line 1027

Reply:
@smottal I did this so it returns earlier so we can "skip" some steps. What you asked was my first approach too.